src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,121 @@ def standardize_patient_name(
+    is the ideal case. If the first attempt fails and a FHIR resource associated with
+    the phone number and a country extraction function are provided we make a second
+    attempt to parse the phone number using any countries extracted from the resource.
+    Finally, in the case where the second attempt fails for a resource and country
+    extraction function are not provided we make a final attempt to parse the number
+    assuming it is American.
+
+    Args:
+        raw (str): Raw phone number to be standardized.
+
+        resource (dict): FHIR resource that the raw phone originated from that possibly
+                         contains information indicating which country the phone number
+                         is from.
+
+        countries (list): A list containing 2 letter ISO country codes for each country
+                          extracted from resource of the phone number to be standardized
+                          that might indicate it the phone numbers country of origin.
+    Returns:
+        standardized (str): The standardized phone number in E.164 format when the raw
+                            phone number was succesfully parsed and an emptry string
+                            otherwise.
     """
 
-    raw = [x for x in raw if x.isnumeric()]
-    raw = "".join(raw)
-    if len(raw) != 10:
-        raw = None
-    return raw
+    if countries != [None, "US"]:
+        countries = [None] + countries + ["US"]
+
+    standardized = ""
+    for country in countries:
+        try:
+            standardized = phonenumbers.parse(raw, country)
+        except phonenumbers.phonenumberutil.NumberParseException:
+            continue
+    if standardized != "":
+        standardized = str(
+            phonenumbers.format_number(
+                standardized, phonenumbers.PhoneNumberFormat.E164
+            )
+        )
+    return standardized
+
+
+def standardize_country(
+    raw_country: str, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
+) -> str:
+    """
+    Given a country return it in a standard form as specified by code_type.
+
+    Args:
+        raw_country (str): Country to be standardized.
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+
+    Returns:
+        standard_country (str): Country in standardized form, or None if unable to
+        standardize.
+    """
+    standard_country = ""
+    if len(raw_country) == 3 and raw_country.isalpha():
+        standard_country = pycountry.countries.get(alpha_3=raw_country)
+    elif len(raw_country) == 2 and raw_country.isalpha():
+        standard_country = pycountry.countries.get(alpha_2=raw_country)
+    elif len(raw_country) == 3 and raw_country.isnumeric():
+        standard_country = pycountry.countries.get(numeric=raw_country)
+    elif len(raw_country) >= 4 and all(x.isalpha() or x.isspace() for x in raw_country):
```

Comment:
Is the `all(...)` condition necessary? What happens with `pycountry.countries.get(name=raw_country)` if it contains non-alpha characters? I'd assume it just wouldn't match anything and so nothing would be returned? I suppose I'd ask that for all of the extra conditions. I just did some testing with `pycountry`, and it seems to handle these scenarios just fine. I think the code would be much cleaner if we did the following, starting on line 158:

```python
raw_country = raw_country.strip().upper()
standard_country = None
if len(raw_country) == 2:
    standard_country = pycountry.countries.get(alpha_2=raw_country)
if len(raw_country) == 3: # no need for elif since the conditions are mutually exclusive
    standard_country = pycountry.countries.get(alpha_3 = raw_country) or pycountry.countries.get(numeric=raw_country)
if len(raw_country) >= 4:
    standard_country = pycountry.countries.get(name=raw_country) or pycountry.countries.get(official_name=raw_country)
    
if standard_country is not None:
    <code>
```

Reply:
nit: I'd also consider dropping the `_country` suffix from the variable name as it should be clear from the function what `raw` and `standard` would represent, and I'd also recommend changing `standard` to `standardized`.

Reply:
That's a carry over from addressing one of Spencer's comments before eliminating the fuzzy matching. Yeah, it isn't needed any more. I'm not inclined to eliminate the use of `elif` though. I agree the conditions are mutually exclusive, but Python doesn't know that and will evaluate all the conditions as opposed to breaking out of the block early after one condition is met.